dvc/command/completion.py

```diff
@@ -0,0 +1,100 @@
+import argparse
+import logging
+import os
+
+import shtab
+
+from dvc.command.base import CmdBaseNoRepo, append_doc_link
+
+logger = logging.getLogger(__name__)
+DEFAULT_FILENAME = {"bash": "dvc", "zsh": "_dvc"}
+CHOICE_FUNCTIONS = {
+    "bash": {"DVCFile": "_dvc_compgen_DVCFiles"},
+    "zsh": {"DVCFile": "_files -g '(*?.dvc|Dvcfile|dvc.yaml)'"},
+}
+PREAMBLE = {
+    "bash": """
+# $1=COMP_WORDS[1]
+_dvc_compgen_DVCFiles() {
+  compgen -d -S '/' -- $1  # recurse into subdirs
+  compgen -f -X '!*?.dvc' -- $1
+  compgen -f -X '!*Dvcfile' -- $1
+  compgen -f -X '!*dvc.yaml' -- $1
+}
+""",
+    "zsh": "",
+}
+
+
+class Optional(shtab.Optional):
+    DVC_FILE = [shtab.Choice("DVCFile", required=False)]
+
+
+class Required(shtab.Required):
+    DVC_FILE = [shtab.Choice("DVCFile", required=True)]
+
+
+class CmdCompletion(CmdBaseNoRepo):
+    def run(self):
+        from dvc.cli import get_main_parser
+
+        parser = get_main_parser()
+        shell = self.args.shell
+        script = shtab.complete(
+            parser,
+            shell=shell,
+            preamble=PREAMBLE[shell],
+            choice_functions=CHOICE_FUNCTIONS[shell],
+        )
+
+        if self.args.dir == "-":
+            logger.debug("Writing tab completion to stdout")
+            print(script)
```

Comment:
I guess we could've used `logger.info` for consistency with other commands. Yeah, I know that `print` is more appropriate in CLI, we've been planning to change up that soon.

Reply:
removed